metaflow/current.py

```diff
@@ -6,28 +6,42 @@ def __init__(self):
         self._run_id = None
         self._step_name = None
         self._task_id = None
+        self._retry_count = None
         self._origin_run_id = None
         self._namespace = None
         self._username = None
         self._is_running = False
 
     def _set_env(self,
-                 flow_name,
-                 run_id,
-                 step_name,
-                 task_id,
-                 origin_run_id,
-                 namespace,
-                 username):
+                 flow_name=None,
+                 run_id=None,
+                 step_name=None,
+                 task_id=None,
+                 retry_count=None,
+                 origin_run_id=None,
+                 namespace=None,
+                 username=None,
+                 is_running=True):
 
         self._flow_name = flow_name
         self._run_id = run_id
         self._step_name = step_name
         self._task_id = task_id
+        self._retry_count = retry_count
         self._origin_run_id = origin_run_id
         self._namespace = namespace
         self._username = username
-        self._is_running = True
+        self._is_running = is_running
+
+    def _update_env(self, env):
+        for k, v in env.items():
+            setattr(self.__class__, k, property(fget=lambda _, v=v: v))
```

Comment:
How is that different from L:46-48? Is that just code duplication or am I missing something?

Reply:
Some weirdness in github maybe.